### PR TITLE
Add prometheus adapter chart

### PR DESCRIPTION
**Pull request by MattiasGees**
**What this PR does / why we need it**:
This PR adds the Prometheus Adapter chart https://github.com/DirectXMan12/k8s-prometheus-adapter. This adapter is needed if you want to do HPA based on custom-metrics from Prometheus.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: incubator/prometheus-adapter/templates/custom-metrics-apiserver-deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    chart: {{ template "k8s-prometheus-adapter.chart" . }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+  name: {{ template "k8s-prometheus-adapter.fullname" . }}
+  namespace: {{ .Release.Namespace | quote }}
+spec:
+  replicas: {{ .Values.replicas }}
+  selector:
+    matchLabels:
+      app: {{ template "k8s-prometheus-adapter.name" . }}
```
Comment (unguiculus) [inline incubator/prometheus-adapter/templates/custom-metrics-apiserver-deployment.yaml]: Add `release` label to selector.

File: incubator/prometheus-adapter/templates/custom-metrics-apiserver-service.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  app: {{ template "k8s-prometheus-adapter.name" . }}
+  chart: {{ template "k8s-prometheus-adapter.chart" . }}
+  release: {{ .Release.Name }}
+  heritage: {{ .Release.Service }}
+spec:
+  ports:
+  - port: 443
+    targetPort: 6443
+  selector:
+    app: {{ template "k8s-prometheus-adapter.name" . }}
```
Comment (unguiculus) [inline incubator/prometheus-adapter/templates/custom-metrics-apiserver-service.yaml]: Add `release` label to selector.

File: incubator/prometheus-adapter/templates/custom-metrics-apiserver-service.yaml [hunk 3]
```diff
@@ -0,0 +1,15 @@
+apiVersion: v1
+kind: Service
+metadata:
+  name: {{ template "k8s-prometheus-adapter.fullname" . }}
+  namespace: {{ .Release.Namespace | quote }}
```
Comment (unguiculus) [inline incubator/prometheus-adapter/templates/custom-metrics-apiserver-service.yaml]: `labels:` missing. Check other places as well.

**Review by unguiculus — CHANGES_REQUESTED** [review]
Please make sure all resources have standard labels. Also, it is not necessary to add the namespace explicitly to the resources.

File: incubator/prometheus-adapter/templates/custom-metrics-apiserver-deployment.yaml (hunk 1 above)
Comment (MattiasGees) [reply incubator/prometheus-adapter/templates/custom-metrics-apiserver-deployment.yaml]: done

File: incubator/prometheus-adapter/templates/custom-metrics-apiserver-service.yaml (hunk 2 above)
Comment (MattiasGees) [reply incubator/prometheus-adapter/templates/custom-metrics-apiserver-service.yaml]: done

**Comment by MattiasGees** [comment]
@unguiculus I addressed your comments.

File: incubator/prometheus-adapter/README.md [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+```console
+$ helm install incubator/prometheus-adapter
+```
+
+## Introduction
+
+This charts install the [Prometheus Adapter](https://github.com/DirectXMan12/k8s-prometheus-adapter) on your kubernetes clutser using the [Helm](https://helm.sh) package manager.
+
+## Prerequisites
+Kubernetes 1.6+
```
Comment (unguiculus) [inline incubator/prometheus-adapter/README.md]: 1.9+

File: incubator/prometheus-adapter/values.yaml [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+  create: true
+
+serviceAccount:
+  # Specifies whether a service account should be created
+  create: true
+  # The name of the service account to use.
+  # If not set and create is true, a name is generated using the fullname template
+  name:
+
+resources:
```
Comment (unguiculus) [inline incubator/prometheus-adapter/values.yaml]: Not used in deployment. Also, please don't specify defaults for resources. Comment them out.

**Review by unguiculus — CHANGES_REQUESTED** [review]

File: incubator/prometheus-adapter/templates/custom-metrics-resource-reader-cluster-role.yaml [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+rules:
+- apiGroups:
+  - ""
+  resources:
+  - namespaces
+  - pods
+  - services
+  - configmaps
```
Comment (florianrusch) [inline incubator/prometheus-adapter/templates/custom-metrics-resource-reader-cluster-role.yaml]: Is there a reason why you add configmaps? In the [original deployment files](https://github.com/DirectXMan12/k8s-prometheus-adapter/blob/master/deploy/manifests/custom-metrics-resource-reader-cluster-role.yaml) are only `namespaces`, `pods` and `services` listed.

File: incubator/prometheus-adapter/values.yaml (hunk 5 above)
Comment (florianrusch) [reply incubator/prometheus-adapter/values.yaml]: If I'm right, it hasn't been used anywhere yet.

**Review by florianrusch — COMMENTED** [review]
Great work :+1:

**Comment by florianrusch** [comment]
I've provided some changes. Take a look here: https://github.com/MattiasGees/charts/pull/1

File: incubator/prometheus-adapter/templates/custom-metrics-resource-reader-cluster-role.yaml (hunk 6 above)
Comment (MattiasGees) [reply incubator/prometheus-adapter/templates/custom-metrics-resource-reader-cluster-role.yaml]: @florianrusch in my testing I ran into the following problem when trying to start it without configmaps.
```
panic: configmaps "extension-apiserver-authentication" is forbidden: User "system:serviceaccount:application-monitoring:prometheus-adapter-prometheus-adapter" cannot get configmaps in the namespace "kube-system": Unknown user "system:serviceaccount:application-monitoring:prometheus-adapter-prometheus-adapter"
```

File: incubator/prometheus-adapter/values.yaml (hunk 5 above)
Comment (MattiasGees) [reply incubator/prometheus-adapter/values.yaml]: This is addressed now.

**Comment by MattiasGees** [comment]
@unguiculus @florianrusch sorry for the long delay but was waiting on the new release of prometheus-adapter that is compatible with the latest k8s. 

I made the necessary changes to work with v0.2.1 of prometheus-adapter and tested it on an k8s 1.10.5 cluster and everything works as expected.

**Comment by unguiculus** [comment]
I'd suggest you move the chart to stable.

**Comment by MattiasGees** [comment]
@unguiculus I moved it to stable.
@florianrusch I think you have to sign the CLA as I merge in some of your commits.

**Comment by florianrusch** [comment]
Hi guys, excuse me. I'm writing the last sentences of my bachelor thesis and haven't had much time in the last days. I'm not very familiar with the CLA stuff. But I hope I did it right a few minutes ago.

**Comment by MattiasGees** [comment]
@florianrusch np, thanks for signing and helping me with the PR.
@unguiculus I think I now need an ok-to-test :)

**Review by unguiculus — APPROVED** [review]

File: stable/prometheus-adapter/templates/custom-metrics-configmap.yaml [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+metadata:
+  name: {{ template "k8s-prometheus-adapter.fullname" . }}
+  labels:
+    app: {{ template "k8s-prometheus-adapter.name" . }}
+    chart: {{ template "k8s-prometheus-adapter.chart" . }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+data:
+  config.yaml: |
+    rules:
```
Comment (ryanolson) [inline stable/prometheus-adapter/templates/custom-metrics-configmap.yaml:12]: It does not appear the rules are configurable.  Am I missing something?

**Review by ryanolson — COMMENTED** [review]
Looks like a good start.

I think adding the flexibility to BYO-rules is important.  I might be missing how to do this.

**Pull request merged**